metricbeat/module/system/socket_summary/socket_summary.go

```diff
@@ -114,17 +124,22 @@ func calculateConnStats(conns []net.ConnectionStat) common.MapStr {
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
-func (m *MetricSet) Fetch(report mb.ReporterV2) {
-
+func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 	// all network connections
 	conns, err := net.Connections("inet")
 
 	if err != nil {
-		report.Error(err)
-		return
+		return errors.Wrap(err, "error getting connections")
 	}
 
+	stats := calculateConnStats(conns)
+	newStats, err := applyEnhancements(stats, m)
+	if err != nil {
+		return errors.Wrap(err, "error applying enhancements")
```

Comment:
I think we could log this and still report the stats we have without the enhancements.

Reply:
Excellent idea!